src/Symfony/Component/HttpKernel/Tests/Controller/ArgumentResolver/QueryOrRequestParameterValueResolverTest.php

```diff
@@ -0,0 +1,292 @@
+                'defaultValue' => false,
+                'isNullable' => false,
+            ],
+            [false], // falls back to false
+        ];
+
+        yield 'parameter not found but nullable' => [
+            [],
+            [
+                'name' => 'firstName',
+                'type' => 'string',
+                'isVariadic' => false,
+                'hasDefaultValue' => false,
+                'defaultValue' => false,
+                'isNullable' => true,
+            ],
+            [],
+        ];
+
+        yield 'parameter not found but optional' => [
+            [],
+            [
+                'name' => 'firstName',
+                'type' => 'string',
+                'isVariadic' => false,
+                'hasDefaultValue' => true,
+                'defaultValue' => false,
+                'isNullable' => false,
+            ],
+            [],
+        ];
+
+        yield 'parameter not found' => [
+            [],
+            [
+                'name' => 'firstName',
+                'type' => 'string',
+                'isVariadic' => false,
+                'hasDefaultValue' => false,
+                'defaultValue' => false,
+                'isNullable' => false,
+            ],
+            '/Missing "(query|request)" parameter "firstName"\.$/',
+        ];
+    }
+
+    /**
+     * @return iterable<string, array{Request, ArgumentMetadata, array<mixed>, ?string}>
+     */
+    public static function queryAndRequestDataProvider(): iterable
+    {
+        foreach (self::provideValidParameters() as $testName => [$parameters, $metadata, $expected]) {
+            yield 'query '.$testName => [
+                Request::create('/', 'GET', $parameters),
+                new ArgumentMetadata(
+                    $metadata['name'],
+                    $metadata['type'],
+                    $metadata['isVariadic'],
+                    $metadata['hasDefaultValue'],
+                    $metadata['defaultValue'],
+                    $metadata['isNullable'],
+                    [new QueryParameter()],
+                ),
+                \is_array($expected) ? $expected : [],
+                \is_string($expected) ? $expected : null,
+            ];
+
+            yield 'request '.$testName => [
+                Request::create('/', 'POST', $parameters),
+                new ArgumentMetadata(
+                    $metadata['name'],
+                    $metadata['type'],
+                    $metadata['isVariadic'],
+                    $metadata['hasDefaultValue'],
+                    $metadata['defaultValue'],
+                    $metadata['isNullable'],
+                    [new RequestParameter()],
+                ),
+                \is_array($expected) ? $expected : [],
+                \is_string($expected) ? $expected : null,
+            ];
+        }
+    }
+
+    /**
+     * @param array<mixed> $expected
+     *
+     * @dataProvider queryAndRequestDataProvider
```

Comment:
please name providers as provide+TestCaseName and put them after their test case

Reply:
Fixed.